packages/aws-cdk-lib/core/lib/helpers-internal/validate-props.ts

```diff
@@ -0,0 +1,42 @@
+import { Construct } from 'constructs';
+import { ValidationError } from '../errors';
+
+/**
+ * Represents a validation rule for props of type T.
+ * @template T The type of the props being validated.
+ */
+export type ValidationRule<T> = {
+  /**
+  * A function that checks if the validation rule condition is met.
+  * @param {T} props - The props to validate.
+  * @returns {boolean} True if the condition is met (i.e., validation fails), false otherwise.
+  */
+  condition: (props: T) => boolean;
+
+  /**
+   * A function that returns an error message if the validation fails.
+   * @param {T} props - The props that failed validation.
+   * @returns {string} The error message.
+   */
+  message: (props: T) => string;
+};
+
+/**
+ * Validates props against a set of rules and throws an error if any validations fail.
+ *
+ * @template T The type of the props being validated.
+ * @param {string} className - The name of the class being validated, used in the error message.
```

Comment:
You should give the example how to use this normally. i.e. `Queue.name`

Reply:
Will do, thanks.